### PR TITLE
chore: add lambda-backed custom resource to NLB partial template

### PR DESCRIPTION
**Pull request by Lou1415926**
This PR:
1. Wait for the certificate to be shown as "unused" before proceeding to delete it
2. Add custom resource to the NLB partial template, as well as its supporting resources.

Preceding PR: #3129

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/template/templates/workloads/partials/cf/nlb.yml [hunk 1]
```diff
@@ -111,4 +110,92 @@ NLBDNSAlias:
...
+    LoadBalancerHostedZoneID: !GetAtt PublicNetworkLoadBalancer.CanonicalHostedZoneID
+    EnvHostedZoneId:
+      Fn::ImportValue:
+        !Sub "${AppName}-${EnvName}-HostedZone"
+    EnvName: !Ref EnvName
+    AppName: !Ref AppName
+    ServiceName: !Ref WorkloadName
+    RootDNSRole: {{ .AppDNSDelegationRole }}
+    DomainName:  {{ .AppDNSName }}
+    Aliases: ["sc-alias-3.ntest.napp.peanutsninja.com", sc-alias-2-sc.ntest.napp.peanutsninja.com]
```
Comment (iamhopaul123) [inline internal/pkg/template/templates/workloads/partials/cf/nlb.yml]: 🤔

File: cf-custom-resources/lib/nlb-cert-validator-updater.js [hunk 2]
```diff
@@ -431,6 +460,10 @@ async function unusedValidationOptions(ownedCertARN, loadBalancerDNS) {
     // Look for validation options that will be no longer needed by this service.
     const certificates = await serviceCertificates();
     const { certOwned: certPendingDeletion, otherCerts: certInUse } = categorizeCertificates(certificates, ownedCertARN);
+    if (!certPendingDeletion) {
+        // Cannot find the certificate that is pending deletion; perhaps it is deleted already. Exit peacefully.
```
Comment (iamhopaul123) [inline cf-custom-resources/lib/nlb-cert-validator-updater.js:474]: If the cert doesn't exist, will it be a problem when trying to delete the cert?

Comment (Lou1415926) [reply cf-custom-resources/lib/nlb-cert-validator-updater.js:474]:
```
 await clients.acm().deleteCertificate({ CertificateArn: certARN }).promise().catch(err => {
        if (err.name !== "ResourceNotFoundException") {
            throw err;
        }
    });
```
This error handling should handle it

File: internal/pkg/template/templates/workloads/partials/cf/nlb.yml [hunk 3]
```diff
@@ -111,4 +110,93 @@ NLBDNSAlias:
         AliasTarget:
           HostedZoneId: !GetAtt PublicNetworkLoadBalancer.CanonicalHostedZoneID
           DNSName: !GetAtt PublicNetworkLoadBalancer.DNSName
+{{- end}}
+{{- if eq .NLB.Listener.Protocol "TLS" }}
+NLBCertManagerAction:
+  Metadata:
+    'aws:copilot:description': "Request and validate the certificate for your Network Load Balancer, as well as adding A-records for the aliases."
```
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/partials/cf/nlb.yml]: nit: to remain within <= 80 chars
```suggestion
    'aws:copilot:description': "Request and validate the certificate for the NLB and add records for aliases"
```

File: internal/pkg/template/templates/workloads/partials/cf/nlb.yml [hunk 4]
```diff
@@ -111,4 +110,93 @@ NLBDNSAlias:
...
+    LoadBalancerHostedZoneID: !GetAtt PublicNetworkLoadBalancer.CanonicalHostedZoneID
+    EnvHostedZoneId:
+      Fn::ImportValue:
+        !Sub "${AppName}-${EnvName}-HostedZone"
+    EnvName: !Ref EnvName
+    AppName: !Ref AppName
+    ServiceName: !Ref WorkloadName
+    RootDNSRole: {{ .AppDNSDelegationRole }}
+    DomainName:  {{ .AppDNSName }}
+    Aliases: {{ .NLB.Listener.Aliases }}
```
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/partials/cf/nlb.yml]: You might need
```suggestion
    Aliases: {{fmtSlice .NLB.Listener.Aliases }}
```

File: internal/pkg/template/templates/workloads/partials/cf/nlb.yml [hunk 5]
```diff
@@ -111,4 +110,93 @@ NLBDNSAlias:
...
+            - Sid: TaggedResourcesRead
+              Effect: Allow
+              Action: tag:GetResources
+              Resource: "*"
+            - Sid: CertificateCreateAndRead
+              Effect: Allow
+              Action:
+                - acm:DescribeCertificate
+                - acm:RequestCertificate
+                - acm:AddTagsToCertificate
```
Comment (efekarakus) [inline internal/pkg/template/templates/workloads/partials/cf/nlb.yml:197]: The permissions look good! Do you know by any chance if for these two actions doing a condition on `aws:RequestTag/${TagKey} ` would narrow it down further?
https://docs.aws.amazon.com/service-authorization/latest/reference/list_awscertificatemanager.html#awscertificatemanager-policy-keys

**Review by efekarakus — COMMENTED** [review]
Looks great!

File: internal/pkg/template/templates/workloads/partials/cf/nlb.yml (hunk 5 above)
Comment (Lou1415926) [reply internal/pkg/template/templates/workloads/partials/cf/nlb.yml:197]: yes! updated!

**Review by efekarakus — APPROVED** [review]

File: cf-custom-resources/lib/nlb-cert-validator-updater.js (hunk 2 above)
Comment (iamhopaul123) [reply cf-custom-resources/lib/nlb-cert-validator-updater.js:474]: How about the describe logic before trying to delete the cert here? https://github.com/aws/copilot-cli/pull/3132/files#diff-904aa8592d00fc56cc75b3a4e5c3d52404d435129b7717885ece3b8e402b1860R251-R258

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**